sdk/storage/azure-storage-queue/azure/storage/queue/aio/_queue_service_client_async.py

```diff
@@ -7,41 +7,37 @@
 
 import functools
 from typing import (
-    Any, Dict, List, Optional, Union,
-    TYPE_CHECKING)
+    Any, Dict, List, Optional,
+    TYPE_CHECKING, Union
+)
+from typing_extensions import Self
 
 from azure.core.async_paging import AsyncItemPaged
 from azure.core.exceptions import HttpResponseError
 from azure.core.pipeline import AsyncPipeline
 from azure.core.tracing.decorator import distributed_trace
 from azure.core.tracing.decorator_async import distributed_trace_async
+from ._models import QueuePropertiesPaged
+from ._queue_client_async import QueueClient
+from .._encryption import StorageEncryptionMixin
+from .._generated.aio import AzureQueueStorage
+from .._generated.models import StorageServiceProperties
+from .._models import QueueProperties, service_properties_deserialize, service_stats_deserialize
+from .._queue_service_client_helpers import _parse_url
 from .._serialize import get_api_version
+from .._shared.base_client import parse_connection_str, StorageAccountHostsMixin
 from .._shared.base_client_async import AsyncStorageAccountHostsMixin, AsyncTransportWrapper
-from .._shared.policies_async import ExponentialRetry
 from .._shared.models import LocationMode
+from .._shared.policies_async import ExponentialRetry
 from .._shared.response_handlers import process_storage_error
-from .._generated.aio import AzureQueueStorage
-from .._generated.models import StorageServiceProperties
-from .._encryption import StorageEncryptionMixin
-from .._models import (
-    service_stats_deserialize,
-    service_properties_deserialize,
-)
-from .._queue_service_client import QueueServiceClient as QueueServiceClientBase
-from ._models import QueuePropertiesPaged
-from ._queue_client_async import QueueClient
 
 if TYPE_CHECKING:
-    from azure.core.credentials import AzureNamedKeyCredential, AzureSasCredential, TokenCredential
-    from .._models import (
-        CorsRule,
-        Metrics,
-        QueueProperties,
-        QueueAnalyticsLogging,
-    )
+    from azure.core.credentials import AzureNamedKeyCredential, AzureSasCredential
+    from azure.core.credentials_async import AsyncTokenCredential
+    from .._models import CorsRule, Metrics, QueueAnalyticsLogging
 
 
-class QueueServiceClient(AsyncStorageAccountHostsMixin, QueueServiceClientBase, StorageEncryptionMixin):
+class QueueServiceClient(AsyncStorageAccountHostsMixin, StorageAccountHostsMixin, StorageEncryptionMixin):  # type: ignore [misc]  # pylint: disable=line-too-long
```

Comment:
This comment applies to both `QueueServiceClientAsync` and `QueueClientAsync`

This is part 1 of why we need to decouple base client, the ignore misc is for:
```
azure\storage\queue\aio\_queue_service_client_async.py:40: error: Definition of "_create_pipeline" in base class "AsyncStorageAccountHostsMixin" is incompatible with definition in base class "StorageAccountHostsMixin"  [misc]
azure\storage\queue\aio\_queue_service_client_async.py:40: error: Definition of "_batch_send" in base class "AsyncStorageAccountHostsMixin" is incompatible with definition in base class "StorageAccountHostsMixin"  [misc]
azure\storage\queue\aio\_queue_service_client_async.py:40: error: Definition of "_format_query_string" in base class "AsyncStorageAccountHostsMixin" is incompatible with definition in base class "StorageAccountHostsMixin"  [misc]
```